server/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshots.java

```diff
@@ -101,9 +106,15 @@ private static SysSnapshot toSysSnapshot(Repository repository, SnapshotId snaps
     }
 
     private static CompletableFuture<SysSnapshot> createSysSnapshot(Repository repository, SnapshotId snapshotId) {
-        return repository.getSnapshotInfo(snapshotId)
-            .thenApply(snapshotInfo -> toSysSnapshot(repository, snapshotId, snapshotInfo))
-            .exceptionally(t -> {
+        return repository.getSnapshotGlobalMetadata(snapshotId).thenCombine(
+            repository.getSnapshotInfo(snapshotId),
+            (metadata, snapshotInfo) -> {
```

Comment:
Just thinking loudly: if getting global metadata is inevitable for now, can we avoid repository.getSnapshotInfo and fetch that info from the global metadata? I think it's out of the hotfix scope anyway but if you think it's possible maybe let's add TODO?

Reply:
Nope, we can't, `SnapshotInfo` has other info like start/end time, state, successful shards, etc.